data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/configuration/ScramAuthConfig.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.kafka.configuration;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+/**
+ * * A helper class that helps to read sasl SCRAM auth configuration values from
+ * pipelines.yaml
+ */
+public class ScramAuthConfig {
+
+    @JsonProperty("username")
+    private String username;
+
+    @JsonProperty("password")
+    private String password;
+
+    @JsonProperty("mechanism")
+    private String mechanism;
```

Comment:
Kafka excepts `SCRAM-SHA-256` and `SCRAM-SHA-512` as mechanisms. Do you think it would be necessary to validate at this point? You will find out pretty much immediately if you enter something invalid so I thought it would be a bit overkill.

Reply:
Thanks @franky-m . Adding validation would help. If you'd like to do this in a follow-on PR that would help.